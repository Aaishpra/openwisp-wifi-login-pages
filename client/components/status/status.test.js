/* eslint-disable prefer-promise-reject-errors */
import axios from "axios";
/* eslint-disable camelcase */
import {shallow} from "enzyme";
import React from "react";
import {Cookies} from "react-cookie";
import ShallowRenderer from "react-test-renderer/shallow";

import getConfig from "../../utils/get-config";
import Status from "./status";

jest.mock("axios");

const defaultConfig = getConfig("default");
const createTestProps = props => {
  return {
    language: "en",
    orgSlug: "default",
    statusPage: defaultConfig.components.status_page,
    cookies: new Cookies(),
    logout: jest.fn(),
    ...props,
  };
};

describe("<Status /> rendering", () => {
  let props;
  it("should render correctly", () => {
    props = createTestProps();
    const renderer = new ShallowRenderer();
    const component = renderer.render(<Status {...props} />);
    expect(component).toMatchSnapshot();
  });
});

describe("<Status /> interactions", () => {
  let props;
  let wrapper;
  it("should call logout function when logout button is clicked", () => {
    axios.mockImplementationOnce(() => {
      return Promise.resolve({});
    });
    props = createTestProps();
    wrapper = shallow(<Status {...props} />);
    wrapper.find("#owisp-status-logout-btn").simulate("click", {});
    expect(wrapper.instance().props.logout.mock.calls.length).toBe(1);
  });
  it("test componentDidMount lifecycle method", () => {
    axios
      .mockImplementationOnce(() => {
        return Promise.reject();
      })
      .mockImplementationOnce(() => {
        return Promise.reject();
      })
      .mockImplementationOnce(() => {
        return Promise.resolve({
          data: {
            "control:Auth-Type": "Accept",
          },
        });
      })
      .mockImplementationOnce(() => {
        return Promise.resolve({
          data: {
            "control:Auth-Type": "Reject",
          },
        });
      });
    props = createTestProps();
    wrapper = shallow(<Status {...props} />);
    return wrapper
      .instance()
      .componentDidMount()
      .then(() => {
        expect(wrapper.instance().props.logout.mock.calls.length).toBe(2);
        return wrapper
          .instance()
          .componentDidMount()
          .then(() => {
            expect(wrapper.instance().props.logout.mock.calls.length).toBe(2);
            return wrapper
              .instance()
              .componentDidMount()
              .then(() => {
                expect(wrapper.instance().props.logout.mock.calls.length).toBe(
                  3,
                );
              });
          });
      });
  });
});
