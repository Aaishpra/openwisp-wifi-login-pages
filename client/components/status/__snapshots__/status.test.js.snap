// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> rendering should render correctly 1`] = `
<React.Fragment>
  <div
    className="owisp-status-container"
  >
    <div
      className="owisp-status-inner"
    >
      <div
        className="owisp-status-content-div"
      >
        <div
          className="owisp-status-content-line"
        >
          WiFi Login Successful!
        </div>
        <div
          className="owisp-status-content-line"
        >
          You can now use the internet.
        </div>
        <div
          className="owisp-status-content-line"
        >
          You may leave this page open in case you want to log out.
        </div>
        <React.Fragment>
          <Link
            className="owisp-status-link"
            to="/default/change-password/"
          >
            Change your password
          </Link>
          <input
            className="owisp-status-btn owisp-status-logout-btn"
            id="owisp-status-logout-btn"
            onClick={[Function]}
            type="button"
            value="Logout"
          />
        </React.Fragment>
      </div>
      <div
        className="owisp-status-contact-div"
      >
        <Memo(Connect(Contact)) />
      </div>
    </div>
  </div>
</React.Fragment>
`;
